boot-speed/testflinger/test_cmds.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+set -ex
+set -o pipefail
+
+# Print some information about the testflinger host system
+cat /etc/os-release
+uname -a
+id
+pwd
+ls -la
+
+# Do not auto-update snaps on the provisioned device for now; we'll
+# update them manually later. If changing this remember that Core
+# may auto-reboot after a snap refresh.
+ssh ubuntu@$DEVICE_IP 'sudo snap set system refresh.hold="$(date --date=tomorrow +%Y-%m-%dT%H:%M:%S%:z)"'
```

Comment:
Is there documentation about the auto-reboot after a snap refresh? This does not seem right. Have you talked to the snapd team?

Part of this testing is to gather the snapd refresh time. By putting a hold on the system refresh are you not preventing us from capturing that data?

Reply:
I didn't read about Core auto rebooting when the core or kernel snaps are refreshed in the official documentation but the thing is well known, and I guess it's written somewhere.

What I'm doing it: measure the boot time before refreshing the snaps, then refresh the snaps (allowing the system to reboot as many times as it wishes), and then measure the boot speed with the updated snaps. The actual refresh time is not recorded. This is different from the time it takes for snapd.seeded.service to complete.

I'm holding the snaps to prevent the system from rebooting while the measurement is being taken.

Measuring the snapd refresh time is not easy. Do we count in the (up to 2) reboots it takes to refresh the snaps? What about the time it takes to download the snaps?

Reply:
The main point of capturing device boot time is to capture the refresh time as a part of the initial boot. Therefore, we need to have a discussion with the snapd team.